src/SDK/Common/Retry/RetryPolicyInterface.php

```diff
@@ -0,0 +1,72 @@
+ * As per the specification (https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/exporter.md#retry),
+ * it is required to implement the exponential retrypolicy with jitter.
+ *
+ * By default there is no retry policy set.
+ *
+ * Here is the example how exporter can use this policy
+ * i.e. lets say if user wants to enable retry policy for grpc exporter
+ *
+ * Add the following code in the exporter;
+ *    Use SpanExporterTrait;
+ *
+ *    The following line will enable the exponential retry policy with jitter with
+ *    default values.
+ *
+ *    $retryPolicy = new ExponentialWithJitterRetryPolicy(
+ *          $maxAttempts = ExponentialWithJitterRetryPolicy::DEFAULT_MAX_ATTEMPTS,
+ *          $initialBackoff = ExponentialWithJitterRetryPolicy::DEFAULT_INITIAL_BACKOFF,
+ *          $maxBackoff = ExponentialWithJitterRetryPolicy::DEFAULT_MAX_BACKOFF,
+ *          $backoffMultiplier = ExponentialWithJitterRetryPolicy::DEFAULT_BACKOFF_MULTIPLIER,
+ *          $jitter = ExponentialWithJitterRetryPolicy::DEFAULT_JITTER,
+ *          $retryableStatusCodes = [
+ *              \GRPC\STATUS_CANCELLED,
+ *              \Grpc\STATUS_DEADLINE_EXCEEDED,
+ *              \Grpc\STATUS_PERMISSION_DENIED,
+ *              \Grpc\STATUS_RESOURCE_EXHAUSTED,
+ *              \Grpc\STATUS_ABORTED,
+ *              \Grpc\STATUS_OUT_OF_RANGE,
+ *              \Grpc\STATUS_UNAVAILABLE,
+ *              \Grpc\STATUS_DATA_LOSS,
+ *              \Grpc\STATUS_UNAUTHENTICATED,
+ *          ],
+ *          $scheduler = null
+ *      );
+ *    $this->setRetryPolicy($retryPolicy);
+ *
+ *    The arguments in ExponentialWithJitterRetryPolicy() are as follows:
+ *          maxAttempts,        - maximum no of retry attempts allowed
+ *          initialBackoff,     - no of secs to wait before retrying for the first time
+ *          maxBackoff,         - maximum no of sec to wait before retry timeout
+ *          backoffMultiplier,  - multiplier value used to calculate the delay before next attempt
+ *          jitter,             - jitter value to add randomness to delay value
+ *          scheduler           - Blocking or Non Blocking delay scheduler (default: BlockingSchuduler)
+ *          retryableStatusCodes- array of retryable status code. All the status codes apart from this will not be retried
+ *      )
+ *
+ *    NonBlockingScheduler will be more useful when fiber is implemented.
+ *    By Default scheduler is of type BlockingScheduler.
+ */
+interface RetryPolicyInterface
+{
+    public function shouldRetry(
+        int $attempt,
+        int $status
+    ): ?bool;
+
+    public function getDelay(int $attempt): int;
+
+    /**
+     * delay the execution of the thread by $timeout milliseconds
+     * @param  int $timeout: milliseconds to delay the execution
+     */
+    public function delay(int $timeout): void;
```

Comment:
> [I have kept it here with the idea that each policy must have a delayschedular](https://github.com/open-telemetry/opentelemetry-php/pull/677#discussion_r919080161)

After working on an async exporter: this should not be part of the retry policy and should be moved to the exporter; providing a scheduler is only useful for exporters that are not explicitely async, async exporters already know the preferred way of delaying execution.

Reply:
I have removed everything out of the retryPolicyInterface.php except then shouldRetry(), delay() and getDelay() function.